File: 2021/10/04/NumberOfLines.java
/**
 * https://leetcode.com/problems/number-of-lines-to-write-string/
 * @author Andrew Jarombek
 * @since 10/4/2021
 */

public class NumberOfLines {
    public int[] numberOfLines(int[] widths, String s) {
        int rows = 1;
        int width = 0;

        for (char c : s.toCharArray()) {
            int i = c - 'a';

            if (widths[i] + width > 100) {
                width = 0;
                rows++;
            }

            width = widths[i] + width;
        }

        return new int[] {rows, width};
    }
}

File: 2022/04/27/numberOfLines.ts
/**
 * https://leetcode.com/problems/number-of-lines-to-write-string/
 * @author Andrew Jarombek
 * @since 4/27/2022
 */

function numberOfLines(widths: number[], s: string): number[] {
    let lines = 1;
    let width = 0;
    let offset = "a".charCodeAt(0);

    for (let c of s) {
        let val = widths[c.charCodeAt(0) - offset];
        width += val;

        if (width > 100) {
            width = val;
            lines++;
        }
    }

    return [lines, width];
}